### DISCUSSION TITLE
Connect multiple ethereum wallets

### DISCUSSION THREAD

**Discussion by vidhathat** [wevm/wagmi#504] [Ideas]
I want to let users add multiple wallets to our website. To be clear it's not a multi-chain feature but a scenario where a user has 2 eth wallets. Currently, wagmi lets users add only a single wallet but we ideally want to let users add more than one wallet from the same chain. Would encourage the community to let me know if this can be possible on wagmi or would truly be grateful if you could point out some other service which does this.

**Comment by tmm** [comment]
You can connect multiple wallets with wagmi currently, but only one can actively listen. This is a relatively uncommon use case and does not have great support across wallets (e.g. MetaMask browser extension can only be connected once per page). Definitely curious if others want this feature to exist and what it would look like.

**Reply by Deveshb15** [reply]
Would love to have this feature, working on an NFT portfolio management tool, so basically I want to allow the user to connect their hot wallet as well as a cold wallet so that he can view all of his NFTs from one centralized place.
Not sure how can this be implemented but just thinking out loud

**Reply by agualis** [reply]
What about this scenario with metamask and safe in the wagmi config:

- Browser tab1: user opens the DApp (foo.dapp.fi) inside `https://app.safe.global/` --> wagmi automatically connects to the Safe account (safe is the active connector)
- Browser tab2: user opens the same Dapp in foo.dapp.fi and connects with metamask --> wagmi connects to the MM account (there are 2 connectors and wagmi is listening to the MM one)
- If you go back to the Browser tab1 you will see that the MM account is the current active one but that is not what we want in the context of app.safe.global
Even if you use `sessionStorage`, that will be the behaviour as the new connections are detected in both tabs. How can we avoid this? Maybe a way to tell wagmi config to always use a given a connector when connected to more than one?

**Comment by allenchuang** [comment]
@tmm This is a desired feature. I believe there will be multiple usecases for this as users would have cold wallets, hot wallets, throwaway wallets..etc. Is there an example somewhere to show how we can connect multiple wallets? thanks!

**Reply by montycheese** [reply]
+1

**Comment by owieth** [comment]
Would love to see a feature like this! 👀

**Comment by oliverox** [comment]
Building a tool with the ability to connect to multiple wallets (just need to read the wallet address). This feature would be really useful. thanks!

**Comment by el-buku** [comment]
any news on this
